### PR TITLE
Added simple shadows sandcastle

### PR DESCRIPTION
**Pull request by lilleyse**
For #2594 

The simple sandcastle demo for shadows. There are a few places where I access scene directly, which I shouldn't do for a Showcase demo. I can maybe make these accessible through `Viewer` instead.

### CODE REVIEW

File: Apps/Sandcastle/gallery/Shadows.html [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+viewer.terrainProvider = new Cesium.CesiumTerrainProvider({
+    url : '//assets.agi.com/stk-terrain/world',
+    requestWaterMask : true,
+    requestVertexNormals : true
+});
+
+var scene = viewer.scene;
+var shadowMap = scene.shadowMap;
+shadowMap._maxmimumDistance = 10000.0;
+scene.globe.castShadows = true;
```
Comment (pjcozzi) [inline Apps/Sandcastle/gallery/Shadows.html]: `globe` is also available off of `viewer`.

File: Apps/Sandcastle/gallery/Shadows.html [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+viewer.terrainProvider = new Cesium.CesiumTerrainProvider({
+    url : '//assets.agi.com/stk-terrain/world',
+    requestWaterMask : true,
+    requestVertexNormals : true
+});
+
+var scene = viewer.scene;
+var shadowMap = scene.shadowMap;
+shadowMap._maxmimumDistance = 10000.0;
```
Comment (pjcozzi) [inline Apps/Sandcastle/gallery/Shadows.html]: We shouldn't access this private member like this.

**Comment by pjcozzi** [comment]
Take a screenshot for this for the Sandcastle gallery.

**Comment by pjcozzi** [comment]
* Add a toggle to turn shadows on/off so users can clearly see them.  We'll also want a toggle for soft shadows and any other common shadows setting.  The idea is that users can come to this sample to learn the shadow properties they need to set.
* Add an option for terrain and primitives to cast/receive shadows
* Change the default view/model/time so that shadows are blatantly obvious.  At least at this time, the shadow on the aircraft is not obvious.

**Comment by lilleyse** [comment]
Updated.

**Comment by pjcozzi** [comment]
When I switch the view from Exton to Half Dome, the aircraft stops animating.  Is that intentional?

**Comment by lilleyse** [comment]
> When I switch the view from Exton to Half Dome, the aircraft stops animating. Is that intentional?

No... I'll look into that.

**Comment by pjcozzi** [comment]
Otherwise, this is great.

Let me know when to merge.

**Comment by mramato** [comment]
Even with a maximumed browser window, I'm getting run in with the standard toolbar:

image

**Comment by lilleyse** [comment]
This should be ready now. I vertically aligned the buttons.

**Pull request merged**